pkg/operator/che_cm.go

```diff
@@ -0,0 +1,133 @@
+		"CHE_WEBSOCKET_ENDPOINT": wsprotocol + "://" + cheHost + "/api/websocket",
+		// todo Make configurable
+		"CHE_DEBUG_SERVER":                              "false",
+		"CHE_INFRASTRUCTURE_ACTIVE":                     infra,
+		"CHE_INFRA_KUBERNETES_BOOTSTRAPPER_BINARY__URL": protocol + "://" + cheHost + "/agent-binaries/linux_amd64/bootstrapper/bootstrapper",
+		"CHE_INFRA_OPENSHIFT_PROJECT":                   workspacesNamespace,
+		"CHE_INFRA_KUBERNETES_PVC_STRATEGY":             pvcStrategy,
+		"CHE_INFRA_KUBERNETES_PVC_QUANTITY":             pvcClaimSize,
+		"CHE_INFRA_OPENSHIFT_TLS__ENABLED":              tls,
+		"CHE_INFRA_KUBERNETES_TRUST__CERTS":             tls,
+		"CHE_JDBC_URL":                                  "jdbc:postgresql://" + postgresHostName + ":" + postgresPort + "/" + chePostgresDb,
+		"CHE_JDBC_USERNAME":                             chePostgresUser,
+		// todo Create a secret for it?
+		"CHE_JDBC_PASSWORD":                             chePostgresPassword,
+		"CHE_LOG_LEVEL":                                 "INFO",
+		"CHE_KEYCLOAK_AUTH__SERVER__URL":                keycloakURL + "/auth",
+		"CHE_INFRA_OPENSHIFT_OAUTH__IDENTITY__PROVIDER": openShiftIdentityProviderId,
+		"CHE_PREDEFINED_STACKS_RELOAD__ON__START":       "true",
+		"CHE_INFRA_KUBERNETES_SERVICE__ACCOUNT__NAME":   "che-workspace",
+		"JAVA_OPTS": "-XX:MaxRAMFraction=2 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 " +
+			"-XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 " +
+			"-XX:AdaptiveSizePolicyWeight=90 -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap " +
+			"-Dsun.zip.disableMemoryMapping=true -Xms20m " + cheWsmasterProxyJavaOptions,
+		"CHE_WORKSPACE_AUTO_START":                              "true",
+		"CHE_KEYCLOAK_REALM":                                    keycloakRealm,
+		"CHE_KEYCLOAK_CLIENT__ID":                               keycloakClientId,
+		"CHE_INFRA_KUBERNETES_WORKSPACE__UNRECOVERABLE__EVENTS": "FailedMount,FailedScheduling,MountVolume.SetUp failed,Failed to pull image",
+		"CHE_WORKSPACE_AGENT_DEV_INACTIVE__STOP__TIMEOUT__MS":   "-1",
+		"CHE_WORKSPACE_JAVA_OPTIONS": "-XX:MaxRAM=150m -XX:MaxRAMFraction=2 -XX:+UseParallelGC " +
+			"-XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 " +
+			"-Dsun.zip.disableMemoryMapping=true " +
+			"-Xms20m -Djava.security.egd=file:/dev/./urandom " + cheWorkspaceProxyJavaOptions,
+		"CHE_WORKSPACE_MAVEN__OPTIONS": "-XX:MaxRAM=150m -XX:MaxRAMFraction=2 -XX:+UseParallelGC " +
+			"-XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 " +
+			"-Dsun.zip.disableMemoryMapping=true " +
+			"-Xms20m -Djava.security.egd=file:/dev/./urandom " + cheWorkspaceProxyJavaOptions,
+		"CHE_WORKSPACE_HTTP__PROXY__JAVA__OPTIONS": cheWorkspaceProxyJavaOptions,
+		"CHE_WORKSPACE_HTTP__PROXY":                cheWorkspaceHttpProxy,
+		"CHE_WORKSPACE_HTTPS__PROXY":               cheWorkspaceHttpsProxy,
```

Comment:
Yes, that is an excellent idea. I was thinking about giving a user possibility to create a configmap that is then either used in Che deployment or merged into an existing che cm. Probably the first option is better since merging/updating brings unnecessary complexity.